authentication_document.py

```diff
@@ -421,7 +421,6 @@ def _update_service_areas(cls, library, areas, type, service_areas):
                 msgs.append(str(_("The following service area was unknown: %(service_area)s.", service_area=json.dumps(unknown))))
             if ambiguous:
                 msgs.append(str(_("The following service area was ambiguous: %(service_area)s.", service_area=json.dumps(ambiguous))))
-            _db.rollback()
```

Comment:
This session-level rollback was interfering with, and redundant with, the subtransaction rollback that now happens when registration of a new library fails for _any_ reason.